// @insurely/eslint-config/react/index.js
module.exports = {
  extends: [
    'eslint-config-airbnb/rules/react',
    'eslint-config-airbnb/rules/react-a11y',
    "plugin:react-hooks/recommended",
  ],
  env: {
    'browser': true,
    'jest': true
  },
  parser: '@babel/eslint-parser',
  rules: {
    'react/destructuring-assignment': ['off'],
    'react/jsx-fragments': ['off'],
    'react/jsx-indent': ['error', 2],
    'react/jsx-indent-props': ['error', 2],
    'react/jsx-one-expression-per-line': 'off',
    'react/jsx-props-no-spreading': 'off',
    'react/prop-types': 'off',
    'react/jsx-no-bind': ['error', {
      'allowArrowFunctions': true,
      'allowFunctions': true
    }],
    "react/jsx-filename-extension": [
      "warn",
      {
        "extensions": [
          ".jsx",
          ".tsx"
        ]
      }
    ],
    "react/require-default-props": "off",
    'react/react-in-jsx-scope': 0,
    "react/function-component-definition": "off",
    "react/no-unstable-nested-components": [
      "error",
     {
       "allowAsProps": true ,
       "customValidators": ["fallbackRender"]
     }
   ]
  },
  settings: {
    polyfills: ['fetch'],
  }
}
