// @insurely/eslint-config/typescript/index.js
module.exports = {
  extends: ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
  parser: "@typescript-eslint/parser",
  plugins: ["@typescript-eslint"],
  rules: {
    "import/extensions": [
      "error",
      "ignorePackages",
      {
        js: "never",
        jsx: "never",
        ts: "never",
        tsx: "never",
      },
    ],
    "no-use-before-define": "off",
    "@typescript-eslint/no-use-before-define": [
      "error"
    ],
    "indent": "off",
    "@typescript-eslint/indent": ["error", 2],
    "no-shadow": "off",
    "@typescript-eslint/no-shadow": "error",
    "no-unused-vars": "off",
    "no-underscore-dangle": "off",
    '@typescript-eslint/no-unused-vars': ['error', { 'argsIgnorePattern': '^_', "varsIgnorePattern": "^_" }],
    '@typescript-eslint/switch-exhaustiveness-check': "error",
  },
  settings: {
    "import/resolver": {
      typescript: {
        project: "./",
      },
      node: {
        extensions: [".js", ".jsx", ".ts", ".tsx"],
        paths: ["./src"],
      },
    },
    "import/extensions": [".js", ".jsx", ".ts", ".tsx"],
    "import/parsers": {
      "@typescript-eslint/parser": [".ts", ".tsx"],
    },
  },
};
