// @insurely/eslint-config/base/index.js
module.exports = {
  extends: 'eslint-config-airbnb/base',
  env: {
    browser: true,
    node: true,
  },
  rules: {
    'import/prefer-default-export': 'off',
    'max-len': ['error', { code: 120 }],
    'max-lines': ['warn', 400],
    'no-multiple-empty-lines': ['error', { max: 1 }],
    'import/order': [
      'error',
      {
        'newlines-between': 'always-and-inside-groups',
        groups: ['builtin', 'external', 'internal', 'parent', 'sibling', 'index'],
        alphabetize: {
          order: 'asc',
          caseInsensitive: true,
        },
      },
    ],
    'import/no-unresolved': [2, { ignore: ['@insurely/ui'] }],
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": [
          "**/*.config.*",
          "**/*.test.*",
          "**/test*",
          "**/vitest*",
          "**/mocks/**"
        ],
        "peerDependencies": true
      }
    ],
    'no-tabs': 0,
    'no-unexpected-multiline': 'error',
    'no-nested-ternary': 0,
    'no-console': 'error',
    'no-unused-vars': ['error', { argsIgnorePattern: '^_', varsIgnorePattern: '^_' }],
    'consistent-return': 0,
  },
};
